.github/workflows/main.yaml

```
@@ -0,0 +1,73 @@
+name: Main CI
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version:
+          - '3.12'
+    steps:
+    - uses: actions/checkout@v4
+    - name: Use Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: 'pip'
+    - name: Install testing dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements-test.txt
+    - name: Lint
+      # FIXME
+      continue-on-error: true
+      run: |
+        flake8 lib/pyld tests --count --show-source --statistics
+      env:
+        LOADER: ${{ matrix.loader }}
+  test:
+    runs-on: ubuntu-latest
+    needs: [lint]
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version:
+          - '3.8'
+          - '3.9'
+          - '3.10'
+          - '3.11'
+          - '3.12'
+          - 'pypy3.10'
+        loader: [requests, aiohttp]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Use Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: 'pip'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Install testing dependencies
+      run: |
+        pip install -r requirements-test.txt
+    - name: Fetch test suites
+      run: |
+        git clone --depth 1 https://github.com/w3c/json-ld-api.git _json-ld-api
+        git clone --depth 1 https://github.com/w3c/json-ld-framing.git _json-ld-framing
+        git clone --depth 1 https://github.com/json-ld/normalization.git _normalization
+    - name: Test with Python=${{ matrix.python-version }} Loader=${{ matrix.loader }}
+      run: |
+        python tests/runtests.py ./_json-ld-api/tests -l ${{ matrix.loader }}
+        python tests/runtests.py ./_json-ld-framing/tests -l ${{ matrix.loader }}
+        python tests/runtests.py ./_normalization/tests -l ${{ matrix.loader }}
```

```suggestion
        python tests/runtests.py -l ${{ matrix.loader }}
```

is going to be sufficient.

---

@anatoly-scherbakov the `submodule` command will have to be added.

---

@BigBlueHat it seems that `actions/checkout` has an option to fetch submodules so we can likely avoid running `make` even.

---

@anatoly-scherbakov @BigBlueHat  ok with you that we make this work and merge this first without the submodule stuff?